hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -2836,182 +2867,97 @@ int MDMParser::socketSend(int socket, const char * buf, int len)
         cnt -= blk;
     }
     LOCK();
-    if (ISSOCKET(socket) && (_sockets[socket].pending == 0) && _atOk()) {
+    if (ISSOCKET(socket) && (_sockets[socket].pending == 0) && _checkModem()) {
         sendFormated("AT+USORD=%d,0\r\n", _sockets[socket].handle); // TCP
-        waitFinalResp(nullptr, nullptr, USORD_TIMEOUT);
+        CHECK_TIMEOUT(waitFinalResp(nullptr, nullptr, USORD_TIMEOUT));
     }
-    UNLOCK();
     return (len - cnt);
-#else
-    int bytesLeft = len;
-    while (bytesLeft > 0) {
-        LOCK();
-        if (!ISSOCKET(socket)) {
-            goto error;
-        }
-        // Maximum number of bytes that can be written via single +USOWR command
-        size_t chunkSize = bytesLeft;
-        if (chunkSize > USO_MAX_WRITE / 2) { // Half the maximum size in binary mode
-            chunkSize = USO_MAX_WRITE / 2;
-        }
-        // Write command prefix
-        char data[256]; // Hex data is written in chunks
-        const int prefixSize = snprintf(data, sizeof(data), "AT+USOWR=%d,%u,\"", _sockets[socket].handle,
-                (unsigned)chunkSize);
-        if (prefixSize < 0 || prefixSize > (int)sizeof(data)) {
-            goto error;
-        }
-        if (send(data, prefixSize) != prefixSize) {
-            goto error;
-        }
-        // Write hex data
-        do {
-            size_t n = chunkSize;
-            if (n > sizeof(data) / 2) {
-                n = sizeof(data) / 2;
-            }
-            bytes2hexbuf_lower_case((const uint8_t*)buf, n, data);
-            const size_t hexSize = n * 2;
-            if (send(data, hexSize) != (int)hexSize) {
-                goto error;
-            }
-            buf += n;
-            bytesLeft -= n;
-            chunkSize -= n;
-        } while (chunkSize > 0);
-        // Write command suffix
-        if (!_atOk()) {
-            goto error;
-        }
-        if (send("\"\r\n", 3) != 3) {
-            goto error;
-        }
-        if (waitFinalResp(nullptr, nullptr, USOWR_TIMEOUT) != RESP_OK) {
-            goto error;
-        }
-        UNLOCK();
-    }
-    return (len - bytesLeft);
-error:
-    UNLOCK();
-    return MDM_SOCKET_ERROR;
-#endif // defined(SOCKET_HEX_MODE)
+}
+
+int MDMParser::_socketError(void) {
+    sendFormated("AT+USOER");
```

Comment:
Gen 2 requires `"\r\n"` trailing most commands.  Without this, the command is not being recognized, and there is no `+USOER: x` URC.